### PR TITLE
Fix English issues in the documentation

### LINKED ISSUE #4635: Documentation is littered with typos
All the typo and grammar issues make a bad first impression on those considering using SCons.

This is a reminder to contribute from [the branch](https://github.com/rico-chet/scons/tree/fix-languagetool-complaints).

### PR DESCRIPTION
**Pull request by rico-chet**
## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` and `RELEASE.txt` (and read the `README.rst`).
* [x] I have updated the appropriate documentation

The change is probably too minor to check any of the boxes above.

Addresses #4635

### CODE REVIEW

**Comment by rico-chet** [comment]
The first commit contains the obvious things which are likely to be OK without a debate. It can be cherry-picked and merged separately to avoid merge conflicts with other branches.

File: SCons/Environment.xml [hunk 1]
```diff
@@ -1033,7 +1033,7 @@ either as separate arguments to the
 &f-Clean;
 method, or as a list.
 &f-Clean;
-will also accept the return value of any of the &consenv;
+will also accept the return value of the &consenv;
```
Comment (mwichmann) [inline SCons/Environment.xml:1036]: A similar change is also in #4648

File: SCons/Environment.xml [hunk 2]
```diff
@@ -1065,7 +1065,7 @@ Clean(['foo', 'bar'], 'something_else_to_clean')
 In this example,
 installing the project creates a subdirectory for the documentation.
 This statement causes the subdirectory to be removed
-if the project is deinstalled.
+if the project is uninstalled.
```
Comment (mwichmann) [inline SCons/Environment.xml:1068]: A similar change is also in #4648

File: SCons/Tool/ninja/ninja.xml [hunk 3]
```diff
@@ -350,15 +350,15 @@ python -m pip install ninja
         <summary>
             <para>
                 Internal value used to specify the function to call with argument env to generate the list of files
-                which if changed would require the &ninja; build file to be regenerated.
+                which -- if changed -- would require the &ninja; build file to be regenerated.
```
Comment (mwichmann) [inline SCons/Tool/ninja/ninja.xml]: not super fond of double dashes as separators, though that may be only a personal bias.

File: doc/python10/summary.xml [hunk 4]
```diff
@@ -40,7 +40,7 @@
     This paper has introduced &SCons;, a next-generation build tool
     with a modular, embeddable architecture and a direct Python
     interface. &SCons; has a global view of the dependencies in a source
-    tree, uses MD5 signatures to decide if derived files are out of date,
```
Comment (mwichmann) [inline doc/python10/summary.xml:43]: this file is "unmaintained" - doesn't hurt to fix this, but it has worse problems.

**Comment by mwichmann** [comment]
In general, this looks good to me, not just fixing typos but improving consistency.  The one question would be whether our esteemed maintainer indeed prefers normalizing on "out-of-date" instead of "out of date" (I'm fine with that change).

File: SCons/Environment.xml (hunk 1 above)
Comment (rico-chet) [reply SCons/Environment.xml:1036]: Feel free to merge #4648 first, I'm OK to resolve the conflicts.

File: SCons/Environment.xml (hunk 2 above)
Comment (rico-chet) [reply SCons/Environment.xml:1068]: Feel free to merge #4648 first, I'm OK to resolve the conflicts.

File: SCons/Tool/ninja/ninja.xml (hunk 3 above)
Comment (rico-chet) [reply SCons/Tool/ninja/ninja.xml]: Can also use parentheses instead if that gets this PR going.

Comment (bdbaddog) [reply SCons/Tool/ninja/ninja.xml]: proper grammar would be `which, if changed, would...`

**Comment by bdbaddog** [comment]
@rico-chet - you should also add a blurb in RELEASE.txt and CHANGES.txt per the checklist at the top.

**Comment by bdbaddog** [comment]
Made a bunch of edits, added CHANGE/RELEASE.txts I'll go ahead an merge.

If you can let me know your actual name we can update the CHANGES.txt blurb for this PR.
Thanks for the great work!

**Comment by rico-chet** [comment]
> If you can let me know your actual name we can update the CHANGES.txt blurb for this PR. Thanks for the great work!

My actual name is _Alex Thiessen_, and it did cost me a few hours of spare time to bring it in. I use SCons at work, find it underappreciated, and am glad to give something back.

**Comment by bdbaddog** [comment]
@rico-chet  - updated CHANGES.txt  - Thanks for your help!

**Pull request merged**